src/Microsoft.Build.UnGAC/Program.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.ExceptionServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.UnGAC
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            try
+            {
+                List<string> allInstancesOfGACUtil = new List<string>();
+                StreamReader output;
+                StreamWriter input;
+
+                Process proc = new Process
+                {
+                    StartInfo = new ProcessStartInfo
+                    {
+                        FileName = "C:\\Windows\\System32\\cmd.exe",
+                        Arguments = "/k", // Keep cmd open until we're done
+                        RedirectStandardOutput = true,
+                        RedirectStandardInput = true,
+                        UseShellExecute = false
+                    }
+                };
+                proc.Start();
+
+                // Hook into standard input/output of the process
+                output = proc.StandardOutput;
+                input = proc.StandardInput;
+
+                if (!input.BaseStream.CanWrite)
+                {
+                    proc.Close();
+                    proc.Dispose();
+                    Console.WriteLine("Could not use where.exe to find gacutil.exe. Exiting...");
+                    return;
+                }
+
+                // Use where.exe to find gacutil.exe
+                input.WriteLine("where /F /R \"C:\\Program Files (x86)\\Microsoft SDKs\\Windows\" gacutil.exe");
+
+                // Throw away the first line - it's the command we just used
+                output.ReadLine();
+                string s = output.ReadLine();
+
+                // Store all instances of gacutil for now.
+                while (!String.IsNullOrEmpty(s))
+                {
+                    allInstancesOfGACUtil.Add(s);
+                    s = output.ReadLine();
+                }
```

Comment:
```suggestion
                // Store all instances of gacutil for now.
                for (string loc = output.ReadLine(); !String.IsNullOrEmpty(loc); loc = output.ReadLine())
                {
                    allInstancesOfGACUtil.Add(s);
                }
```

Does this terminate appropriately? If so, I think there's a ReadAllLines function that might do the same. (I might have made that up 😄.)

Also, do we really care about any gacutils after the first?

Reply:
> Does this terminate appropriately? If so, I think there's a ReadAllLines function that might do the same. (I might have made that up 😄.)
It does and has always, which kind of worries me. I would do ReadAllLines but have experienced lockups from that as well. Perhaps I can clear the buffer (as I do below) so I can safely pull everything.

> Also, do we really care about any gacutils after the first?
Honestly I don't think so. I did it because "why not" while I had them there. The final version will almost certainly not store all instances.